config/paths.js

```diff
@@ -1,13 +1,303 @@
+import { Stores } from 'models/stores';
+import { StoreProducts } from 'models/storeProducts';
+import { Suppliers } from 'models/suppliers';
+import { SupplierProducts } from 'models/supplierProducts';
+import config from 'config';
 export const paths = [
     {
-        path: '/assign-roles',
+        path: '/assign-roles/',
         scopes: [SCOPE_TYPE.SUPER_ADMIN],
         method: 'PUT'
     },
     {
-        path: '/roles',
-        scopes: [SCOPE_TYPE.ADMIN, SCOPE_TYPE.SUPER_ADMIN],
+        path: '/roles/',
+        scopes: [SCOPE_TYPE.SUPER_ADMIN],
+        method: 'POST'
+    },
+    {
+        //Ask if scope is correct or not
+        path: '/stores/',
+        scopes: [SCOPE_TYPE.SUPER_ADMIN],
         method: 'POST'
+    },
+    {
+        path: '/stores/',
+        scopes: [SCOPE_TYPE.SUPER_ADMIN, SCOPE_TYPE.STORE_ADMIN],
+        method: 'GET',
+        authMiddleware: async (req, res, next) => {
+            const requesterEmail = req.user[`${config().apiAudience}/email`];
+
+            const configObj = {
+                condition: { 'admin.email': requesterEmail },
+                ownerKey: 'admin',
+                findAllKey: 'admin.email',
+                findAllValue: null, // will be added in the middleware
+                resourceOwnershipPath: 'admin[0].email' // will be used to get the findAllValue from the resource
+            };
+            return await checkOwnership(requesterEmail, Stores, configObj);
+        }
+    },
+    {
+        path: '/stores/:_id',
+        scopes: [SCOPE_TYPE.SUPER_ADMIN, SCOPE_TYPE.STORE_ADMIN],
+        method: 'GET',
+        authMiddleware: async (req, res, next) => {
+            const requesterEmail = req.user[`${config().apiAudience}/email`];
+            const configObj = {
+                condition: { _id: req.params._id },
+                ownerKey: 'admin'
+            };
+            return await checkOwnership(requesterEmail, Stores, configObj);
+        }
+    },
+    {
+        path: '/stores/:_id',
+        scopes: [SCOPE_TYPE.SUPER_ADMIN, SCOPE_TYPE.STORE_ADMIN],
+        method: 'PATCH',
+        authMiddleware: async (req, res, next) => {
+            const requesterEmail = req.user[`${config().apiAudience}/email`];
+            const configObj = {
+                condition: { _id: req.params._id },
+                ownerKey: 'admin'
+            };
+            return await checkOwnership(requesterEmail, Stores, configObj);
+        }
+    },
+
+    {
+        path: '/stores/:_id',
+        scopes: [SCOPE_TYPE.SUPER_ADMIN, SCOPE_TYPE.STORE_ADMIN],
+        method: 'DELETE',
+        authMiddleware: async (req, res, next) => {
+            const requesterEmail = req.user[`${config().apiAudience}/email`];
+            const configObj = {
+                condition: { _id: req.params._id },
+                ownerKey: 'admin'
+            };
+            return await checkOwnership(requesterEmail, Stores, configObj);
+        }
+    },
```

Comment:
we can reuse the authMiddleware function for the above 3. The way I see it the only thing that changes is condition,  ownerKey and models. Let's delete to another function so that all we pass is these 3 things. It should pretty much cover all our cases

Reply:
Ok, I will make another function for it.